src/common/error.ts

```diff
@@ -29,6 +29,35 @@ export class UploadError extends Error {
   }
 }
 
+export class NoTokenListed extends Error {
+  /** @internal */
+  constructor() {
+    super(`LIST ERROR: you must list at least one token.`);
+  }
+}
+
+export class NotOwner extends Error {
+  /** @internal */
+  constructor() {
+    super(`LIST ERROR: you should be the owner of the token to list it.`);
+  }
+}
+
+export class NotAuthorised extends Error {
+  /** @internal */
+  constructor() {
+    super(
+      `LIST ERROR: you do not have the permission to list in the marketplace`,
+    );
+  }
+}
+export class BuyLimit extends Error {
+  /** @internal */
+  constructor(message: string) {
+    super(`BUY ERROR: You cannot buy more tgan ${message} tokens`);
```

Comment:
`tgan` is a typo

Reply:
and in this case, `message` should not be the quantity, otherwise you'd name it `quantity`.

Here's the errors I'm thinking about:


- `MissingRoleError` -> Throw this when someone should have a role, but is missing that role. It should accept 2 parameters:
    - `address` the address thats missing the permission
    - `role` the role that the address is missing
- `AssetNotFoundError` -> The asset you're trying to use could not be found. If we're passing in an asset contract address in the market list method, we should check to see if it exists, if not, throw this error and include the asset address in the error message
- `NotEnoughTokensError` -> Anywhere someone tries to perform an action where they're trying to use more tokens than they have, throw this error. Parameters include:
    - The address of the assets contract
    - The quantity that was requested
    - Optionally, the amount available (which should be less than needed, otherwise this error would not be thrown)